OpenRA.Mods.Common/Lint/CheckSequences.cs

```diff
@@ -19,15 +19,71 @@ class CheckSequences : ILintPass
 	{
 		public void Run(Action<string> emitError, Action<string> emitWarning, Map map)
 		{
-			var sequences = MiniYaml.MergeLiberal(map.SequenceDefinitions, Game.ModData.Manifest.Sequences.Select(s => MiniYaml.FromFile(s)).Aggregate(MiniYaml.MergeLiberal));
+			var sequenceDefinitions = MiniYaml.MergeLiberal(map.SequenceDefinitions,
+				Game.ModData.Manifest.Sequences.Select(s => MiniYaml.FromFile(s)).Aggregate(MiniYaml.MergeLiberal));
 
 			foreach (var actorInfo in map.Rules.Actors)
+			{
 				foreach (var renderInfo in actorInfo.Value.Traits.WithInterface<RenderSimpleInfo>())
 				{
 					var image = renderInfo.Image ?? actorInfo.Value.Name;
-					if (!sequences.Any(s => s.Key == image.ToLowerInvariant()) && !actorInfo.Value.Name.Contains("^"))
+					if (!actorInfo.Value.Name.Contains("^") && !sequenceDefinitions.Any(s => s.Key == image.ToLowerInvariant()))
 						emitWarning("Sprite image {0} from actor {1} has no sequence definition.".F(image, actorInfo.Value.Name));
 				}
+
+				foreach (var traitInfo in actorInfo.Value.Traits.WithInterface<ITraitInfo>())
```

Comment:
This has turned into a monster.  I suggest extracting some of the inner loops / blocks into their own methods.